var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -374,14 +407,67 @@ def cmake_args(self):
                 '-DTPL_ENABLE_SuperLU:BOOL=OFF',
             ])
 
-        # python
-        if '+python' in spec:
+        if '+pnetcdf' in spec:
+            options.extend([
+                '-DTPL_ENABLE_Pnetcdf:BOOL=ON',
+                '-DTPL_Netcdf_Enables_Netcdf4:BOOL=ON',
+                '-DTPL_Netcdf_PARALLEL:BOOL=ON',
+                '-DPNetCDF_ROOT:PATH=%s' % spec['parallel-netcdf'].prefix
+            ])
+        else:
+            options.extend([
+                '-DTPL_ENABLE_Pnetcdf:BOOL=OFF'
+            ])
+
+        if '+zlib' in spec:
             options.extend([
-                '-DTrilinos_ENABLE_PyTrilinos:BOOL=ON'
+                '-DTPL_ENABLE_Zlib:BOOL=ON',
+                '-DZlib_ROOT:PATH=%s' % spec['zlib'].prefix,
```

Comment:
i would specify as much as possible to avoid picking up system libraries instead.